transports/quic/tests/smoke.rs

```diff
@@ -68,614 +31,411 @@ async fn async_std_smoke() {
-impl RequestResponseCodec for PingCodec {
-    type Protocol = PingProtocol;
-    type Request = Ping;
-    type Response = Pong;
-
-    async fn read_request<T>(&mut self, _: &PingProtocol, io: &mut T) -> io::Result<Self::Request>
-    where
-        T: AsyncRead + Unpin + Send,
-    {
-        upgrade::read_length_prefixed(io, 4096 * 10)
-            .map(|res| match res {
-                Err(e) => Err(io::Error::new(io::ErrorKind::InvalidData, e)),
-                Ok(vec) if vec.is_empty() => Err(io::ErrorKind::UnexpectedEof.into()),
-                Ok(vec) => Ok(Ping(vec)),
-            })
-            .await
-    }
+async fn smoke<P: Provider>() {
+    let _ = env_logger::try_init();
 
-    async fn read_response<T>(&mut self, _: &PingProtocol, io: &mut T) -> io::Result<Self::Response>
-    where
-        T: AsyncRead + Unpin + Send,
-    {
-        upgrade::read_length_prefixed(io, 4096 * 10)
-            .map(|res| match res {
-                Err(e) => Err(io::Error::new(io::ErrorKind::InvalidData, e)),
-                Ok(vec) if vec.is_empty() => Err(io::ErrorKind::UnexpectedEof.into()),
-                Ok(vec) => Ok(Pong(vec)),
-            })
-            .await
-    }
+    let (a_peer_id, mut a_transport) = create_transport::<P>();
+    let (b_peer_id, mut b_transport) = create_transport::<P>();
 
-    async fn write_request<T>(
-        &mut self,
-        _: &PingProtocol,
-        io: &mut T,
-        Ping(data): Ping,
-    ) -> io::Result<()>
-    where
-        T: AsyncWrite + Unpin + Send,
-    {
-        upgrade::write_length_prefixed(io, data).await?;
-        io.close().await?;
-        Ok(())
-    }
+    let addr = start_listening(&mut a_transport, "/ip4/127.0.0.1/udp/0/quic").await;
+    let ((a_connected, _, _), (b_connected, _)) =
+        connect(&mut a_transport, &mut b_transport, addr).await;
 
-    async fn write_response<T>(
-        &mut self,
-        _: &PingProtocol,
-        io: &mut T,
-        Pong(data): Pong,
-    ) -> io::Result<()>
-    where
-        T: AsyncWrite + Unpin + Send,
-    {
-        upgrade::write_length_prefixed(io, data).await?;
-        io.close().await?;
-        Ok(())
-    }
+    assert_eq!(a_connected, b_peer_id);
+    assert_eq!(b_connected, a_peer_id);
 }
 
-#[cfg(feature = "async-std")]
-#[async_std::test]
-async fn dial_failure() {
-    let _ = env_logger::try_init();
-    let mut a = create_swarm::<quic::async_std::Provider>().await;
-    let mut b = create_swarm::<quic::async_std::Provider>().await;
+fn generate_tls_keypair() -> libp2p::identity::Keypair {
+    libp2p::identity::Keypair::generate_ed25519()
+}
 
-    let addr = start_listening(&mut a, "/ip4/127.0.0.1/udp/0/quic").await;
+fn create_transport<P: Provider>() -> (PeerId, Boxed<(PeerId, StreamMuxerBox)>) {
+    let keypair = generate_tls_keypair();
+    let peer_id = keypair.public().to_peer_id();
+    let mut config = quic::Config::new(&keypair);
+    config.handshake_timeout = Duration::from_secs(1);
```

Comment:
```suggestion
```
Timeout of 1s is too short when I run it locally. It triggers a `HandshakeTimeout` for the `concurrent_connections_and_streams_async_std` test. Tests pass for me with the default value, which is 5s.

Reply:
Changed it to use the default config!